app/Http/Requests/Sharing/SetSharingRequest.php

```diff
@@ -21,7 +23,7 @@ class SetSharingRequest extends BaseApiRequest implements HasAlbumIDs, HasUserID
 	 */
 	public function authorize(): bool
 	{
-		return AccessControl::can_upload();
+		return Gate::check(UserPolicy::UPLOAD, User::class);
```

Comment:
Here is something weird. Normally, the 2nd parameter is the model onto which the requested action should be applied. For example, `Gate::check(AlbumPolicy::EDIT, $album)` checks if the currently authenticated user is allowed to edit the given album. But this kind of interpretation does not make sense here. We don't want to check whether the currently authenticated user is allowed to upload a user.